### ISSUE TITLE
`ContinuousIntegrationBuild` breaks `__SOURCE_DIRECTORY__`

### ISSUE THREAD

**Issue by MangelMaxime** [dotnet/fsharp#17288]
Hello, 

having `ContinuousIntegrationBuild` set to `true` breaks `__SOURCE_DIRECTORY__` .

**Repro steps**

Provide the steps required to reproduce the problem:

1. Create a folder `Minimal` and inside it:
1. Create `Main.fs` with

	```fs
	printfn "%A" __SOURCE_DIRECTORY__
	```

2. Create `Repro.fsproj` with

	```xml
	<Project Sdk="Microsoft.NET.Sdk">
	    <PropertyGroup>
	        <OutputType>Exe</OutputType>
	        <TargetFramework>net8.0</TargetFramework>
	        <GenerateDocumentationFile>true</GenerateDocumentationFile>
	        <ContinuousIntegrationBuild>true</ContinuousIntegrationBuild>
	    </PropertyGroup>
	    <ItemGroup>
	        <Compile Include="Main.fs" />
	    </ItemGroup>
	</Project>
	```

3. Run `dotnet run` and see the reported path be something like `/_//Minimal"`

If you remove `ContinuousIntegrationBuild` then the reported path is correct.

If possible attach a zip file with the repro case. This often makes it easier for others to reproduce.
The zip file should ideally represent the situation just before the call/step that is problematic.

[Reproduction.zip](https://github.com/user-attachments/files/15685699/Reproduction.zip)

**Expected behavior**

Should report the absolute path even if `ContinuousIntegrationBuild` is set

**Actual behavior**

Report `/_/` instead of the actual absolute path

**Known workarounds**

None

**Related information**

Provide any related information (optional):

* Operating system: Mac OS / Githubs Action Ubuntu
* .NET Runtime kind (.NET Core, .NET Framework, Mono): .NET Core
* Editing Tools (e.g. Visual Studio Version, Visual Studio)

**Comment by baronfel** [comment]
This is just a side effect of the way SourceMaps are written to PDBs and stuff - IIRC changing this would break PDBs, tooling just has to be aware of this and adjust for it. We've had to do it in Ionide, VS handles it, etc. IIRC this comes mostly from the [pathmap](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/advanced) concept, and it something that is required to be compatible with other tooling.

**Comment by MangelMaxime** [comment]
I indeed read something about `pathmap` on this issue https://github.com/dotnet/fsharp/issues/9379

I am not sure if there something to fix or not.

This bug is not too impactful in my situation, as it surfaced because I tried to use [EasyBuild.FileSystemProvider](https://github.com/easybuild-org/EasyBuild.FileSystemProvider) in a build project (similar to a MakeFile). So I could disable `ContinuousIntegrationBuild` just for the `Build.fsproj` if needed.
